tests/test_enforce_generator.py

```
@@ -629,5 +629,189 @@ def test_credential_leak_template_fails_closed_on_shape_drift(self):
+            "  hook: .claude/hooks/auto/test.sh\n"
+            "  generated_from: memory/p5.md\n"
+            "  language: sh\n"
+            "  template: block-on-match-guard.sh.template\n"
+            "---\nBody.\n"
+        )
+        hook_source = generate_hook(md, template_dir=TEMPLATE_DIR)
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".sh", delete=False) as f:
+            f.write(hook_source)
+            tmp_path = f.name
+        try:
+            result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)